pkg/apis/application/v1alpha1/applicationset_types.go

```diff
@@ -196,11 +200,24 @@ type ApplicationSetGenerator struct {
 	Merge                   *MergeGenerator       `json:"merge,omitempty" protobuf:"bytes,8,name=merge"`
 
 	// Selector allows to post-filter all generator.
-	Selector *metav1.LabelSelector `json:"selector,omitempty" protobuf:"bytes,9,name=selector"`
+	Selector *ApplicationLabelSelector `json:"selector,omitempty" protobuf:"bytes,9,name=selector"`
 
 	Plugin *PluginGenerator `json:"plugin,omitempty" protobuf:"bytes,10,name=plugin"`
 }
 
+// +structType=atomic
+type ApplicationLabelSelector struct {
```

Comment:
Maybe juste LabelSelector ? As it's not meant to select only Applications

Reply:
Yes you're right, I was thinking about `LabelSelector\